collectors/proc.plugin/proc_net_wireless.c

```diff
@@ -0,0 +1,453 @@
+
+                rrdset_flag_set(wireless_dev->st_discarded_packets, RRDSET_FLAG_DETAIL);
+
+                wireless_dev->rd_nwid = rrddim_add(wireless_dev->st_discarded_packets, "nwid", NULL, 1,
+                                                   1, RRD_ALGORITHM_INCREMENTAL);
+                wireless_dev->rd_crypt = rrddim_add(wireless_dev->st_discarded_packets, "crypt", NULL, 1,
+                                                    1, RRD_ALGORITHM_INCREMENTAL);
+                wireless_dev->rd_frag = rrddim_add(wireless_dev->st_discarded_packets, "frag", NULL, 1,
+                                                   1, RRD_ALGORITHM_INCREMENTAL);
+                wireless_dev->rd_retry = rrddim_add(wireless_dev->st_discarded_packets, "retry", NULL, 1,
+                                                    1, RRD_ALGORITHM_INCREMENTAL);
+                wireless_dev->rd_misc = rrddim_add(wireless_dev->st_discarded_packets, "misc", NULL, 1,
+                                                   1, RRD_ALGORITHM_INCREMENTAL);
+            } else {
+                rrdset_next(wireless_dev->st_discarded_packets);
+            }
+
+            rrddim_set_by_pointer(wireless_dev->st_discarded_packets, wireless_dev->rd_nwid,
+                                  (collected_number)wireless_dev->nwid);
+
+            rrddim_set_by_pointer(wireless_dev->st_discarded_packets, wireless_dev->rd_crypt,
+                                  (collected_number)wireless_dev->crypt);
+
+            rrddim_set_by_pointer(wireless_dev->st_discarded_packets, wireless_dev->rd_frag,
+                                  (collected_number)wireless_dev->frag);
+
+            rrddim_set_by_pointer(wireless_dev->st_discarded_packets, wireless_dev->rd_retry,
+                                  (collected_number)wireless_dev->retry);
+
+            rrddim_set_by_pointer(wireless_dev->st_discarded_packets, wireless_dev->rd_misc,
+                                  (collected_number)wireless_dev->misc);
+
+            rrdset_done(wireless_dev->st_discarded_packets);
+        }
+
+        if (likely(do_beacon)) {
+            wireless_dev->missed_beacon = str2kernel_uint_t(procfile_lineword(ff, l, 10));
+
+            if (unlikely(!wireless_dev->st_missed_beacon)) {
+                wireless_dev->st_missed_beacon = rrdset_create_localhost("wireless",
+                                                                         wireless_dev->chart_id_net_missed_beacon,
+                                                                         NULL,
+                                                                         wireless_dev->name,
+                                                                         "wireless.missed",
+                                                                         "Number of periodic beacons from the Cell or the Access Point we have missed.",
+                                                                         "packets/s",
```

Comment:
i am not sure it is _from the Cell or the Access Point_, i suggest to name it just _Number of missed beacons._

> packets/s

they are not packets, they are frames (data link layer).

Reply:
So it is in the docs. I meant a server could be AP/client that is why _from the Cell or the Access Point_ looks wrong for me.

Reply:
btw context: `wireless.missed_beacons`